dss/index/bundle.py

```diff
@@ -64,14 +67,14 @@ def _read_file_infos(cls, replica: Replica, fqid: BundleFQID, manifest: JSON) ->
                 try:
                     file_string = handle.get(bucket_name, file_blob_key).decode("utf-8")
                 except BlobStoreError as ex:
-                    if attempts < 5:
+                    # Requeue at the end, yielding to other files we may not have tried yet
+                    files_info.append((file_info, attempts + 1))
+                    if wait_time + len(files_info) * load_time < indexer.remaining_time:
```

Comment:
Good point. Indexer depends on Bundle but Bundle should not depend on Indexer. In fact, if I add a `Indexer` type hint to this method, I'd have to import `Indexer` and that would result in a circular import.

However, this code *does* need to know the remaining run time on a continuing basis. One alternative to passing an `Indexer` instance is passing the lambda context directly. I thought that that was uglier because it exposes Bundle to something even further removed from its core concern.

The reason I had made peace with this ugliness is that it is going to go away once we either ensure that blobs are copied in order (#1130) or pursue @xbrianh's idea of retrying the entire indexer lambda using the @amork's DLQ mechanism.